src/Context/cart-context.js

```diff
@@ -0,0 +1,18 @@
+import {createContext, useContext, useReducer} from "react"
+import { products } from "../backend/db/products"
+import { cartReducer } from "./cart-reducer"
+
+const CartContext = createContext()
+
+const CartProvider = ({children}) => {
+
+  const [state, dispatch] = useReducer(cartReducer, {
+    products: products,
+    cart: []
+  })
+  return <CartContext.Provider value={{state, dispatch}}>{children}</CartContext.Provider>
+}
+
+export const useCart = () => useContext(CartContext)
+
+export default CartProvider
```

Comment:
I think CartContextProvider would be better name but CartProvider also looks fine

Reply:
sure will change that